### PR TITLE
fix broken config validation for required parameters

### PR DESCRIPTION
**Pull request by J-N-K**
The `ConfigDescriptionValidator` implementation did not iterate over the content of the `ConfigDescription` but over the content of the provided configuration. As a result parameters that were missing (and not just set to `null`) are not detected.

Tests results are re-ordered due to fixed typos.

Signed-off-by: Jan N. Klug <github@klug.nrw>

### CODE REVIEW

File: bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorTest.java [hunk 1]
```diff
@@ -241,7 +245,20 @@ public void assertValidationThrowsExceptionContainingMessagesForAllRequiredConfi
...
+        List<ConfigValidationMessage> expected = List.of(new ConfigValidationMessage(parameterName,
+                MessageKey.PARAMETER_REQUIRED.defaultMessage, MessageKey.PARAMETER_REQUIRED.key));
+
+        try {
+            params.remove(parameterName);
+            configDescriptionValidator.validate(params, CONFIG_DESCRIPTION_URI);
+            failBecauseOfMissingConfigValidationException();
+        } catch (ConfigValidationException e) {
+            assertThat(getConfigValidationMessages(e), is(expected));
+        }
```
Comment (wborn) [inline bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorTest.java:258]: JUnit supports checking expected exceptions with `Assertions.assertThrows` like this nowadays:

```suggestion
        params.remove(parameterName);
        ConfigValidationException exception = assertThrows(ConfigValidationException.class,
                () -> configDescriptionValidator.validate(params, CONFIG_DESCRIPTION_URI));
        assertThat(getConfigValidationMessages(exception), is(expected));
```

**Review by wborn — COMMENTED** [review]
Thanks for fixing the bug! I have one small comment below.

File: bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorTest.java (hunk 1 above)
Comment (J-N-K) [reply bundles/org.openhab.core.config.core/src/test/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorTest.java]: Done. Also for all other occurrences in this file.

**Review by wborn — APPROVED** [review]
Thanks!

**Pull request merged**